### DISCUSSION TITLE
Detect unpremium user

### DISCUSSION THREAD

**Discussion by mehmetpeker** [akshaaatt/Google-IAP#116] [Q&A] [answered]
How can I detect if a user has not buy any product or subscription purchases?The functions 'onSubscriptionRestored' or 'onProductRestored' are not triggered if the user has not buy any purchase.Please add this callback to the library in future versions.

**Comment by jdleung** [comment, marked as answer]
It might a wrong way, but that's a solution I can find:
1.  Set `addBillingClientConnectionListener` to check billing connection when app starts.
2. If `status` is `true` in `onConnected`, set all subscription and nonconsume products to `false`.
3. Then `onSubscriptionRestored` will be called if product is valid, just set the product to `true` again.

Important: be sure that the connection `status` is true before setting a product to `false`
